components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -157,6 +165,22 @@ open class GleanInternalAPI internal constructor () {
         logger.info("Registering pings for ${pings.javaClass.canonicalName}")
     }
 
+    /**
+     * Register a closure that will be called once Glean has been initialized.  If Glean is already
+     * initialized, then the callback will be executed immediately. This can be useful if using
+     * Glean from a library where Glean may not be initialized when the library needs to record
+     * metrics or interact with a part of the Glean API that requires it to be initialized.
+     *
+     * @param callback A function/closure that takes no parameters and returns nothing (Unit).
+     */
+    internal fun queueTaskForInit(callback: () -> Unit) {
```

Comment:
I think that if we move the logic to `Dispatchers`, this might not be needed anymore.

Reply:
This was just an artifact of starting this out with this function as part of the public API.  If we move the queue to Dispatchers.kt, then this is not needed here at all.